appengine-java21/datastore-indexes-perfect/src/main/webapp/WEB-INF/appengine-web.xml

```diff
@@ -15,6 +15,11 @@
   limitations under the License.
 -->
 <appengine-web-app xmlns="http://appengine.google.com/ns/1.0">
-  <runtime>java8</runtime>
-  <threadsafe>true</threadsafe>
+  <runtime>java21</runtime>
+  <app-engine-apis>true</app-engine-apis>
```

Comment:
might want to add a comment that this enables legacy bundled apis if possible.

Reply:
No. This is generic change, fully documented in doc pages.
https://cloud.google.com/appengine/docs/standard/java-gen2/services/access